### PR TITLE
build: Fix compilation issue for non-linux platform

### PR DESCRIPTION
**Pull request by sayboras**
### Description

This commit is to make sure there is no compilation issue
with windows for netlink and netns packages

Relates to https://github.com/vishvananda/netlink/issues/146
Relates to https://github.com/vishvananda/netns/issues/23

For #16843

Signed-off-by: Tam Mach <tam.mach@cilium.io>
Suggested-by: Tobias Klauser <tobias@cilium.io>

### CODE REVIEW

File: pkg/defaults/defaults.go [hunk 1]
```diff
@@ -15,7 +13,7 @@ const (
 
 	// AddressScopeMax controls the maximum address scope for addresses to be
 	// considered local ones with HOST_ID in the ipcache
-	AddressScopeMax = int(netlink.SCOPE_LINK) - 1
+	AddressScopeMax = 0xfd - 1 // int(netlink.SCOPE_LINK) - 1
```
Comment (tklauser) [inline pkg/defaults/defaults.go]: Could we move this definition to a `defaults_linux.go` file in the same package, so we can retain the usage of the predefined `netlink.SCOPE_LINK` constant?

File: pkg/ip/ip_scope_unspecified.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+//go:build !linux
+
+package ip
+
+import (
+	"fmt"
+)
+
+var ErrNotImplemented = fmt.Errorf("not implemented")
```
Comment (tklauser) [inline pkg/ip/ip_scope_unspecified.go]: Nit:
```suggestion
var ErrNotImplemented = errors.New("not implemented")
```

**Review by tklauser — APPROVED** [review]

**Review by aditighag — APPROVED** [review]
Thanks! Did you test the changes, or do you want someone else to run the ginkgo tests on mac OS?

**Comment by aditighag** [comment]
Tested the changes on macOS - 

```
$ ginkgo -dryRun
level=info msg="environment variable \"K8S_VERSION\" was not set; setting to default value \"1.23\""
Running Suite: Suite-
=====================
Random Seed: 1651503513
Will run 0 of 0 specs


Ran 0 of 0 Specs in 0.000 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**Comment by sayboras** [comment]
Thanks a lot :100:

**Pull request merged**